www/generate_tutorial/src/tutorial.roc

```diff
@@ -0,0 +1,113 @@
+app "roc-tutorial"
+    packages { pf: "../../../examples/static-site-gen/platform/main.roc" }    
+    imports [
+        pf.Html.{ html, head, body, footer, script, div, main, p, section, h1, h2, label, ol, input, text, nav, a, li, link, meta },
+        pf.Html.Attributes.{ content, name, for, id, type, href, rel, lang, class, title, charset, src },
+    ]
+    provides [transformFileContent] to pf
+
+transformFileContent : Str, Str -> Str
+transformFileContent = \_, htmlContent ->
+    Html.render (view htmlContent)
+
+view : Str -> Html.Node
+view = \htmlContent ->
+    html [lang "en"] [
+        head [] [
+            meta [charset "utf-8"] [],
+            Html.title [] [text "Roc Tutorial"],
+            meta [name "description", content "Learn how to use the Roc programming language."] [],
+            meta [name "viewport", content "width=device-width"] [],
+            link [rel "stylesheet", href "/site.css"] [],
+            link [rel "icon", href "/favicon.svg"] [],
+        ],
+        body [] [
+            viewNavbar,
+            main [] [
+                viewTutorialStart,
+                text htmlContent,
+            ],
+            footer [] [
+                text "Made by people who like to make nice things. © 2022"
+            ]
+        ],
+        script [src "/site.js"] [],
+    ]
+
+viewNavbar : Html.Node
+viewNavbar =
+    div [id "top-bar"] [
+        nav [] [
+            a [class "home-link", href "/", title "The Roc Programming Language"] [text "roc"],
+            div [id "top-bar-links"] [
+                a [href "/tutorial"] [text "tutorial"],
+                a [href "/install"] [text "install"],
+                a [href "/repl"] [text "repl"],
+                a [href "/builtins/Bool"] [text "docs"],
+            ],
+        ],
+    ]
+
+viewTutorialStart : Html.Node
+viewTutorialStart =
+    div [id "tutorial-start"] [
+        input [id "tutorial-toc-toggle", name "tutorial-toc-toggle", type "checkbox"] [],
+        nav [id "tutorial-toc"] [
+            label [id "close-tutorial-toc", for "tutorial-toc-toggle"] [text "close"],
+            # TODO fix search: input [id "toc-search", type "text", placeholder "Search"] [],
```

Comment:
The search function does not work on the current version of the tutorial and I could not find any code supporting the search functionality so I commented it out for now.